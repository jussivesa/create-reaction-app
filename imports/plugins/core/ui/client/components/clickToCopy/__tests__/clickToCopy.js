/**
 * Mock Translation component import, as it uses Meteor modules we have a hard time testing with Jest
 */
jest.mock("/imports/plugins/core/ui/client/components", () => ({
  Translation(props) {
      return <span>{props.defaultValue}</span>; // eslint-disable-line
  }
}));

import React from "react";
import { shallow } from "enzyme";
import shallowToJSON from "enzyme-to-json";
import ClickToCopy from "../clickToCopy";

/**
 * ClickToCopy is a display element that will copy text to your clipboard when clicked
 * It recieves props, and acts accordingly
 */

afterEach(() => {
  jest.clearAllMocks();
});

/**
 * Snapshots make sure your UI does not change unexpectedly
 */

test("ClickToCopy snapshot test", () => {
  const component = shallow((
    <ClickToCopy
      copyToClipboard="Text to copy to clipboard"
      displayText="Text to display"
      i18nKeyTooltip="path.to.i18n.key"
      tooltip="Tooltip text"
    />
  ));
  const tree = shallowToJSON(component);
  expect(tree).toMatchSnapshot();
});
