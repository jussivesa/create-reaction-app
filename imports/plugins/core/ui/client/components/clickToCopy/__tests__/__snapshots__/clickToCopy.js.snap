// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickToCopy snapshot test 1`] = `
<span
  className="rui"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  style={
    Object {
      "display": "inline-flex",
    }
  }
>
  <undefined
    attachement="middle left"
    tooltipContent={null}
  >
    <CopyToClipboard
      onCopy={[Function]}
      text="Text to copy to clipboard"
    >
      <span>
        Text to display
      </span>
    </CopyToClipboard>
  </undefined>
</span>
`;
